hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperTask.java

```diff
@@ -120,14 +145,42 @@ public Pair<String, Boolean> reprocess(OMMetadataManager omMetadataManager) {
       return new ImmutablePair<>(getTaskName(), false);
     }
     try {
-      writeToTheDB(containerKeyMap, containerKeyCountMap, deletedKeyCountList);
+      // only batch write containerKeyCountMap to the containerKeyCountTable
+      // containerKeyMap has been written to the DB
+      // deleted container list is not needed since "reprocess" only has
+      // put operations
+      writeToTheDB(Collections.emptyMap(), containerKeyCountMap,
+          Collections.emptyList());
     } catch (IOException e) {
       LOG.error("Unable to write Container Key Prefix data in Recon DB.", e);
       return new ImmutablePair<>(getTaskName(), false);
     }
     return new ImmutablePair<>(getTaskName(), true);
   }
 
+  private boolean flushAndCommitContainerKeyMapToDB(
+      Map<ContainerKeyPrefix, Integer> containerKeyMap) {
+    try {
+      writeToTheDB(containerKeyMap, Collections.emptyMap(),
```

Comment:
I think, we should flush the "`containerKeyCountMap`" also and optimize more memory footprint if we are introducing this flush mechanism, so you can refactor the code a bit and pass "`containerKeyCountMap`" instead of sending empty map in 2nd argument.

Reply:
Thank you for the suggestion. I have included `containerKeyCountMap`  into the flush, and removing the last `writeToDB` in the 'reprocess' function. I have also moved the flush function to an outer scope to prevent inaccurate container key info when there are multiple bucket layouts.

Reply:
Thanks @ivandika3 for fixing comments. Patch LGTM. +1